eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -180,146 +195,156 @@ jobs:
+        customBuildArgs="--ci --prepareMachine"
+        if [[ '${{ parameters.runOnline }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --online"
+        fi
 
         if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
-          poisonArg='--poison'
+          customBuildArgs="$customBuildArgs --poison"
         fi
 
-        docker run --rm $dockerVolumeArgs -w /vmr $dockerEnvArgs ${{ parameters.container }} ./build.sh --source-only $poisonArg --test $(additionalBuildArgs) /p:SmokeTestConsoleVerbosity=detailed
-      displayName: Run Tests
-
-  # Don't use CopyFiles@2 as it encounters permissions issues because it indexes all files in the source directory graph.
-  - script: |
-      set -x
-
-      targetFolder=$(Build.StagingDirectory)/BuildLogs/
-      mkdir -p ${targetFolder}
-
-      cd "$(sourcesPath)"
-      find artifacts/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find artifacts/prebuilt-report/ -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find src/ -type f -name "*.log" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "*.binlog" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.diff" -exec cp {} --parents -t ${targetFolder} \;
-      find test/ -type f -name "Updated*.txt" -exec cp {} --parents -t ${targetFolder} \;
-    displayName: Prepare BuildLogs staging directory
-    continueOnError: true
-    condition: succeededOrFailed()
+        if [[ '${{ parameters.buildFromArchive }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-repository https://github.com/dotnet/dotnet"
+          customBuildArgs="$customBuildArgs --source-version $(git -C "${{ parameters.vmrPath }}" rev-parse HEAD)"
+        fi
+
+        if [[ '${{ parameters.buildSourceOnly }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --source-only"
+        fi
+
+        if [[ '${{ parameters.useMonoRuntime }}' == 'True' ]]; then
+          customBuildArgs="$customBuildArgs --use-mono-runtime"
+        fi
+
+        if [[ -n "${{ parameters.container }}" ]]; then
+          useDocker=true
+        fi
+
+        if [[ ! -z '${{ parameters.targetRid }}' ]]; then
+          extraBuildProperties="--"
+          if [[ '${{ parameters.useMonoRuntime }}' == 'True' ]]; then
+              customEnvVars="$customEnvVars CROSSCOMPILE=1"
+          fi
+          extraBuildProperties="$extraBuildProperties /p:PortableBuild=true /p:DotNetBuildVertical=true /p:CrossBuild=true"
+        fi
+
+        if [[ ! -z '${{ parameters.crossRootFs }}' ]]; then
+          customEnvVars="$customEnvVars ROOTFS_DIR=${{ parameters.crossRootFs}}"
+        fi
+
+        if [[ ! -z '${{ parameters.targetRid }}' ]]; then
```

Comment:
Can this be combined with the previous `if [[ ! -z '${{ parameters.targetRid }}' ]];` body at https://github.com/dotnet/installer/pull/18409/files#diff-dc549eeb99aa73960e74c9d4c160a3c76e3e414398bb348c1d5365aac0369ad5R275?